src/WebJobs.Script.WebHost/Security/Authorization/Policies/AuthorizationOptionsExtensions.cs

```diff
@@ -49,6 +50,32 @@ public static void AddScriptPolicies(this AuthorizationOptions options)
                     return false;
                 });
             });
+
+            options.AddPolicy(PolicyNames.SystemKeyAuthLevel, p =>
+            {
+                p.AddScriptAuthenticationSchemes();
+                p.RequireAssertion(c =>
+                {
+                    if (c.Resource is AuthorizationFilterContext filterContext)
+                    {
+                        if (filterContext.HttpContext.Request.IsAppServiceInternalRequest())
+                        {
+                            return true;
+                        }
+
+                        HttpContext context = filterContext.HttpContext;
+                        string name = context.Request.GetRequestUri().AbsolutePath.ToLower().Split("/")[3];
```

Comment:
This is pretty tightly coupled to the URI template used by the extensions endpoint. We should look at the route data and that that value instead. We can also change the convention to use something like `keyname` (or `extensionname` as a fallback) for the route data value.